pkg/steps/source.go

```diff
@@ -41,33 +41,61 @@ const (
 	gopath        = "/go"
 	sshPrivateKey = "/sshprivatekey"
 	sshConfig     = "/ssh_config"
+	oauthToken    = "/oauth-token"
+
+	OauthSecretKey = "oauth-token"
 )
 
+type CloneAuthType string
+
+var (
+	CloneAuthTypeSSH   CloneAuthType = "SSH"
+	CloneAuthTypeOAuth CloneAuthType = "OAuth"
+)
+
+type CloneAuthConfig struct {
+	Secret *coreapi.Secret
+	Type   CloneAuthType
+}
+
+func (c *CloneAuthConfig) determineCloneURI(org, repo string) string {
+	if c.Type == CloneAuthTypeSSH {
+		return fmt.Sprintf("ssh://git@github.com/%s/%s.git", org, repo)
+	}
+	return fmt.Sprintf("https://github.com/%s/%s.git", org, repo)
+}
+
 var (
 	JobSpecAnnotation = fmt.Sprintf("%s/%s", CiAnnotationPrefix, "job-spec")
 )
 
-func sourceDockerfile(fromTag api.PipelineImageStreamTagReference, workingDir string, sshSecretName string) string {
+func sourceDockerfile(fromTag api.PipelineImageStreamTagReference, workingDir string, cloneAuthConfig *CloneAuthConfig) string {
 	var dockerCommands []string
 	dockerCommands = append(dockerCommands, "")
 	dockerCommands = append(dockerCommands, fmt.Sprintf("FROM %s:%s", api.PipelineImageStream, fromTag))
 	dockerCommands = append(dockerCommands, "ADD ./app.binary /clonerefs")
 
-	if len(sshSecretName) > 0 {
+	if cloneAuthConfig != nil && cloneAuthConfig.Type == CloneAuthTypeSSH {
 		dockerCommands = append(dockerCommands, fmt.Sprintf("ADD %s /etc/ssh/ssh_config", sshConfig))
 		dockerCommands = append(dockerCommands, fmt.Sprintf("COPY ./%s %s", coreapi.SSHAuthPrivateKey, sshPrivateKey))
 	}
 
+	if cloneAuthConfig != nil && cloneAuthConfig.Type == CloneAuthTypeOAuth {
+		dockerCommands = append(dockerCommands, fmt.Sprintf("COPY ./%s %s", OauthSecretKey, oauthToken))
+	}
```

Comment:
nit: I think that something like this would better convey the actual logic. Pls consider.

```go
if cloneAuthConfig != nil {
  switch cloneAuthConfig.Type {
  case CloneAuthTypeSSH:
    ...
  case CloneAuthTypeOAuth:
    ...
  }
}
```